src/internal/connector/onedrive/collections.go

```diff
@@ -377,13 +385,30 @@ func (c *Collections) Get(
 				c.statusUpdater,
 				c.ctrl,
 				CollectionScopeUnknown,
-				true)
+				true,
+				nil)
 			if err != nil {
 				return nil, false, clues.Wrap(err, "making collection").WithClues(ictx)
 			}
 
 			c.CollectionMap[driveID][fldID] = col
 		}
+
+		numDriveItems := c.NumItems - numPrevItems
+		numPrevItems = c.NumItems
+
+		// Only create a drive cache if there are less than 300k items in the drive.
+		if numDriveItems < urlCacheDriveItemThreshold {
+			logger.Ctx(ictx).Info("adding url cache for drive ", driveID)
```

Comment:
Please avoid adding dynamic values to the log.  DriveID should already be embedded in the ctx.
```suggestion
			logger.Ctx(ictx).Info("adding url cache for drive")
```